Smartway.UiComponent/Inputs/NumericDateEntry.xaml.cs

```diff
@@ -0,0 +1,120 @@
+            BindableProperty.Create(nameof(ErrorCommand), typeof(ICommand), typeof(NumericDateEntry), null);
+
+
+        public int GlobalFontSize
+        {
+            get => (int)GetValue(GlobalFontSizeProperty);
+            set => SetValue(GlobalFontSizeProperty, value);
+        }
+
+        public DateTime FilledDateTime 
+        {
+            get => (DateTime)GetValue(FilledDateTimeProperty);
+            set => SetValue(FilledDateTimeProperty, value);
+        }
+
+        public ICommand ErrorCommand
+        {
+            get => (ICommand)GetValue(ErrorCommandProperty);
+            set => SetValue(ErrorCommandProperty, value);
+        }
+
+        private List<Entry> _dateEntries;
+        public NumericDateEntry()
+        {
+            InitializeComponent();
+
+            _dateEntries = new List<Entry> {
+                FirstEntry,
+                SecondEntry,
+                ThirdEntry
+            };
+            SetFilledDate();
+            SetEventsOnDateEntries();
+        }
+
+        private void SetFilledDate()
+        {
+            //Find how to treat nullable datetime
+            if (FilledDateTime.Year == 1 )
+                return;
+
+            var filledDateTime = FilledDateTime;
+            FirstEntry.Text = filledDateTime.Day.ToString();
+            SecondEntry.Text = filledDateTime.Month.ToString();
+            ThirdEntry.Text = filledDateTime.Year.ToString().Substring(2);
+        }
+
+        private void SetEventsOnDateEntries()
+        {
+            _dateEntries.ForEach(_ => _.PropertyChanged += EntryOnPropertyChanged);
+        }
+
+        private void EntryOnPropertyChanged(object sender, EventArgs e)
+        {
+            var currentEntry = sender as Entry;
+            if (IsValidDateEntry(currentEntry))
+                return;
+
+            var hasFocusEmptyEntry = FocusNextEmptyEntry();
+            if (hasFocusEmptyEntry) 
+                return;
+
+            CheckDateIsComplete();
+        }
+
+        private bool FocusNextEmptyEntry()
+        {
+            foreach (var entry in _dateEntries)
+            {
+                if (IsValidDateEntry(entry))
+                    continue;
+
+                entry.Focus();
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool IsValidDateEntry(Entry entry)
+        {
+            return !(string.IsNullOrEmpty(entry.Text) || entry.Text.Length < 2);
+        }
+
+        private void CheckDateIsComplete()
+        {
+            var day = int.Parse(FirstEntry.Text);
+            var month = int.Parse(SecondEntry.Text);
+            var year = int.Parse(20+ThirdEntry.Text); // hack for add 20 to 21 to make 2021 <3 incoming bug in 2099
+
+            try
+            {
+                FilledDateTime = new DateTime(year, month, day);
+            }
+            catch (Exception exception)
+            {
+                if(ErrorCommand != null)
```

Comment:
can we invert if please?

Reply:
Surtout écrire ErrorCommand?.Execute(exception);